src/core/namespace.c

```diff
@@ -287,10 +288,15 @@ static int apply_mount(
 
                 /* First, get rid of everything that is below if there
                  * is anything... Then, overmount it with an
-                 * inaccessible directory. */
+                 * inaccessible path. */
                 umount_recursive(m->path, 0);
 
-                what = "/run/systemd/inaccessible";
+                r = lstat(m->path, &inaccessible_target);
```

Comment:
I don't know what's the best: add complicated code similar to [EvaluateSymlinksInsideApp](https://github.com/coreos/rkt/blob/master/stage1/init/common/pod.go#L757) inside systemd, or avoid the problem by having two directives `InaccessibleDirectories` and `InaccessibleFiles`.